src/components/common/FolderButton.tsx

```diff
@@ -15,7 +15,10 @@ export default function FolderButton({ isFolderChosen, folderInfo }: FolderButto
 
   function handleSetFolder(e: React.MouseEvent<HTMLElement>) {
     const { currentTarget } = e;
-    setSelectedFolder({ id: currentTarget.dataset.key, name: currentTarget.dataset.name });
+
+    const id = +(currentTarget.dataset.key ?? 0);
+    const name = currentTarget.dataset.name ?? '';
+    setSelectedFolder({ id, name });
```

Comment:
일반적으로 DOM요소(또는 DOM구조) 의존하는 것은 좋지않습니다. 
기능(도메인계층)보다는 UI(표현계층)가 더 유연하고(그래야하고) UI만 변하는 경우도 많겠죠. 
의존하는 방향은 언제나 변하기 쉬운 것에서 변하기 어려운, 변하지 않는 것으로의 방향인 것이 건전합니다

그리고 이 경우에는, 위에 이야기를 무시하고 단순하게 접근해도, 아래 표현이 더 알기 쉽게 표현하는 방법일 거라 생각합니다

```suggestion
    setSelectedFolder({ id: folderInfo.id, name: folderInfo.name });
```

근데 어느정도 이상은 이미 알고계실거같고, dataset 사용해보신 건 좋아요! 👍